back/app/services/phase_insights_service.rb

```diff
@@ -0,0 +1,298 @@
+class PhaseInsightsService
+  SUPPORTED_CATEGORICAL_FIELD_TYPES = %w[select checkbox multiselect].freeze
+
+  def insights_data(phase)
+    cached_insights_data(phase)
+  end
+
+  private
+
+  def initialize
+    @permissions_custom_fields_service = Permissions::PermissionsCustomFieldsService.new
+  end
```

Comment:
Would probably initialize this using `||=` syntax instead of in the `initialize`:

```suggestion
  def permissions_custom_fields_service 
    @permissions_custom_fields_service ||= Permissions::PermissionsCustomFieldsService.new
  end
```